src/async-wrap.h

```diff
@@ -36,27 +36,29 @@ namespace node {
 
 #define NODE_ASYNC_PROVIDER_TYPES(V)                                          \
   V(NONE)                                                                     \
-  V(CRYPTO)                                                                   \
+  V(CONNECTION)                                                               \
   V(FSEVENTWRAP)                                                              \
   V(FSREQWRAP)                                                                \
   V(GETADDRINFOREQWRAP)                                                       \
   V(GETNAMEINFOREQWRAP)                                                       \
   V(HTTPPARSER)                                                               \
   V(JSSTREAM)                                                                 \
-  V(PIPEWRAP)                                                                 \
+  V(PBKDF2REQUEST)                                                            \
   V(PIPECONNECTWRAP)                                                          \
+  V(PIPEWRAP)                                                                 \
   V(PROCESSWRAP)                                                              \
   V(QUERYWRAP)                                                                \
+  V(RANDOMBYTESREQUEST)                                                       \
   V(SHUTDOWNWRAP)                                                             \
   V(SIGNALWRAP)                                                               \
   V(STATWATCHER)                                                              \
-  V(TCPWRAP)                                                                  \
   V(TCPCONNECTWRAP)                                                           \
+  V(TCPWRAP)                                                                  \
   V(TIMERWRAP)                                                                \
   V(TLSWRAP)                                                                  \
   V(TTYWRAP)                                                                  \
-  V(UDPWRAP)                                                                  \
   V(UDPSENDWRAP)                                                              \
+  V(UDPWRAP)                                                                  \
   V(WRITEWRAP)                                                                \
   V(ZLIB)
```

Comment:
`AsyncWrap` uses an `enum` so it can store and more easily access the `char*` of each provider's name for `provider_type()`, and a couple other things. There will be a native API for async hooks, but I'm not including that in this PR. After this lands I'll post the native API PR.